undertow/src/main/java/org/wildfly/extension/undertow/ApplicationSecurityDomainDefinition.java

```diff
@@ -335,8 +335,13 @@ protected RemoveHandler(AbstractAddStepHandler addOperation) {
         protected void performRemove(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
             if (context.isResourceServiceRestartAllowed()) {
                 String securityDomainName = context.getCurrentAddressValue();
-                context.removeService(new SingleSignOnManagerServiceNameProvider(securityDomainName).getServiceName());
-                context.removeService(new SingleSignOnSessionFactoryServiceNameProvider(securityDomainName).getServiceName());
+                context.addStep(new OperationStepHandler() {
+                    @Override
+                    public void execute(OperationContext operationContext, ModelNode modelNode) throws OperationFailedException {
+                        context.removeService(new SingleSignOnManagerServiceNameProvider(securityDomainName).getServiceName());
+                        context.removeService(new SingleSignOnSessionFactoryServiceNameProvider(securityDomainName).getServiceName());
+                    }
+                }, OperationContext.Stage.RUNTIME);
```

Comment:
This was my fault - but I don't think we should be removing services from this method.  Instead move these 2 lines to an overridden performRuntime(...), which should additionally calls the super implementation, so that we still remove the service normally removed by ServiceRemoveStepHandler.